packages/text-input/src/TextInput.tsx

```diff
@@ -39,6 +39,16 @@ const Mode = {
 
 type Mode = typeof Mode[keyof typeof Mode];
 
+export const SizeVariantType = {
+  XSmall: 'xsmall',
+  Small: 'small',
+  Default: 'default',
+  Medium: 'medium',
+  Large: 'large',
+} as const;
+
+export type SizeVariantType = typeof SizeVariantType[keyof typeof SizeVariantType];
```

Comment:
I think we should just call this `Size` to keep in line with e.g. `State`, and other components that have a `State` variant